### ISSUE TITLE
vllm_tt: presence_penalty, frequency_penalty, repetition_penalty silently ignored — not applied on device or CPU

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#3331]
## Issue

When a user sets `presence_penalty`, `frequency_penalty`, or `repetition_penalty` in `SamplingParams`, the vLLM TT plugin accepts the values but silently discards them. Token selection is done from completely unpenalized logits. The parameters have zero effect.

This is a P0 regression relative to the GPU path, which applies all three penalties correctly via `apply_all_penalties` before sampling.

## Root Cause

Three gaps in `vllm_tt/metadata.py` and `vllm_tt/sampler.py`, inherited from the upstream TPU sampler:

1. **`no_penalties` hardcoded `True`** — `XLASupportedSamplingMetadata.no_penalties` was never set to `False`. Because this bool is captured at `torch.compile(dynamic=False)` trace time, the penalty branch was completely absent from every compiled TTIR graph regardless of user-provided values.

2. **Penalty fields never populated** — `frequency_penalties`, `presence_penalties`, `repetition_penalties` were untyped class-level `None` vars (not dataclass fields). `from_input_batch` never read them from `InputBatch` or transferred them to device.

3. **No `apply_penalties` in the sampler** — `vllm_tt/sampler.py` (forked from `vllm/v1/sample/tpu/sampler.py`) had no penalty math. There is also no CPU fallback; `apply_penalties` exists only in the GPU sampler paths and is never called from the TT/TPU model runner.

## Repro

Capture TTIR from a penalty test with and without the fix:

```bash
# WITH fix — penalty test produces a distinct 8th compiled graph with 30 extra ops
TTXLA_LOGGER_LEVEL=DEBUG pytest tests/integrations/vllm_plugin/sampling/test_sampling_params.py \
  -m "nightly and single_device" -k "test_sampling_param_sweep and presence_penalty" -v -s \
  2>&1 | tee penalty_withfix.log
python3 scripts/extract_mlir_graphs.py penalty_withfix.log --type ttir --output-dir /tmp/penalty_withfix

# WITHOUT fix — penalty test produces identical IR to no-penalty baseline
git checkout <pre-fix SHA> -- integrations/vllm_plugin/vllm_tt/sampler.py \
  integrations/vllm_plugin/vllm_tt/metadata.py integrations/vllm_plugin/vllm_tt/model_runner.py
TTXLA_LOGGER_LEVEL=DEBUG pytest tests/integrations/vllm_plugin/sampling/test_sampling_params.py \
  -m "nightly and single_device" -k "test_sampling_param_sweep and presence_penalty" -v -s \
  2>&1 | tee penalty_nofix.log
python3 scripts/extract_mlir_graphs.py penalty_nofix.log --type ttir --output-dir /tmp/penalty_nofix
```

Results from `single_device` OPT-125m:

| Run | Test | Fix | Graphs | Sampling graph ops |
|-----|------|-----|:------:|:-----------------:|
| 1 | temperature (no penalty) | ✅ | 7 | 123 |
| 2 | presence_penalty | ✅ | **8** | **153** |
| 3 | presence_penalty | ❌ | 7 | **123** ← identical to baseline |

Run 3's sampling graph is byte-identical to Run 1 (same MD5). The 30 extra ops in Run 2 (`ttir.gt`, `ttir.where`, `ttir.multiply`, `ttir.subtract`, `ttir.reshape`, `ttir.broadcast`) correspond exactly to the penalty math.

> **Note on existing tests**: `test_sampling_param_sweep[presence_penalty]` passes even without the fix because it uses `temperature=0.8` and only asserts `min_unique >= 2` across 3 outputs — random sampling variation is sufficient to pass that bar. It does not verify that penalty values actually change logits.

## Suggested Fix

PR branch: `kmabee/sampling_params_round2`, commit `7b1de9c7b`.

- `vllm_tt/metadata.py`: promote penalty fields to typed dataclass fields; add `output_token_counts` tensor; add `_compute_token_counts` helper; set `no_penalties=False` in `from_input_batch` when any penalty is non-default.
- `vllm_tt/sampler.py`: add `apply_penalties` (element-wise masking only — no sort/scatter/gather, consistent with `apply_top_k_top_p`). Call it in `sample()` before temperature scaling. Penalty order matches GPU reference: repetition → frequency → presence.
- `vllm_tt/model_runner.py`: pass `vocab_size=self.vocab_size` at both `from_input_batch` call sites so `output_token_counts` is populated.

**Known limitation**: `output_token_counts` counts only generated output tokens. The GPU reference applies repetition penalty to prompt tokens as well. Prompt token support requires additional infrastructure and is tracked separately.
